web/src/engine/websites/Baozimh.ts

```diff
@@ -46,18 +46,21 @@ export default class extends DecoratableMangaScraper {
     }
 
     public override async FetchPages(chapter: Chapter): Promise<Page[]> {
-        const pagesList : Page[] = [];
-        let uri = new URL(chapter.Identifier, this.URI);
-        const sectionSlot = uri.searchParams.get('section_slot');
-        const chapterSlot = uri.searchParams.get('chapter_slot');
-        const linkRegex = new RegExpSafe(`/${sectionSlot}_${chapterSlot}_\\d+\\.html?$`, 'i');
-        for (let run = true; run;) {
-            const data = await FetchHTML(new Request(uri));
-            const pages = [...data.querySelectorAll('.comic-contain amp-img.comic-contain__item')];
-            pagesList.push(...pages.map(element => new Page(this, chapter, new URL(element.getAttribute('src'), this.URI))));
-            uri = new URL([...data.querySelectorAll<HTMLAnchorElement>('div.comic-chapter div.next_chapter a')].at(-1)?.pathname, this.URI);
-            run = linkRegex.test(uri.href);
+        const pagesList: Page[] = [];
+        let chapterPageUri = new URL(chapter.Identifier, this.URI);
+        const sectionSlot = chapterPageUri.searchParams.get('section_slot');
+        const chapterSlot = chapterPageUri.searchParams.get('chapter_slot');
+        for (let hasNextPage = true, pageIndex = 2; hasNextPage; pageIndex++) {
```

Comment:
I don't see a need for this loop, all images could be extracted from a [single HTML](https://www.czmanga.com/comic/chapter/wotianmingdafanpai-tianmingfanpaiyuanzhuhuizhudongman/0_53.html) file (which can be acquired by following the redirect of the chapter link).

Reply:
Did you see the "1/3"? A page only display 50 images max. In the past we had complains that is was multipaged and it doesnt seems to have changed.

If you check the [first page](https://www.czmanga.com/comic/chapter/wotianmingdafanpai-tianmingfanpaiyuanzhuhuizhudongman/0_53.html) and the [second page](https://www.czmanga.com/comic/chapter/wotianmingdafanpai-tianmingfanpaiyuanzhuhuizhudongman/0_53_2.html)  you will see that the pictures are not the same and that second page follow the first.

Reply:
No i didn't see the other pages due to their bad website design. I guess as an average user I would have missed almost two 3rd of the chapters content.

Reply:
We first missed it when website was added ;)